app/controllers/registrations_controller.rb

```diff
@@ -1,18 +1,26 @@
-class Contractors::RegistrationsController < Devise::RegistrationsController
+class RegistrationsController < Devise::RegistrationsController
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  def new
+    build_resource(role: params[:role])
+    yield resource if block_given?
+    respond_with resource
```

Comment:
I would like you to try another way of doing this:

```ruby
def new
  super { | resource | resource.role = params[:role] }
end
```
See if it works. This is a better way to overwrite parts of a method. This is possible because Device has the line `yield resource if block_given?` in the original method.